src/transactions.rs

```diff
@@ -0,0 +1,456 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Seahorse library.
+
+// This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+// You should have received a copy of the GNU General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! The assets module.
+//!
+//! This module defines [Transaction], [TransactionEditor], and [Transactions], which provide CURD (create, read,
+//! update, and delete) operations, with the use of [KeyValueStore] to control the assets resource.
+
+use crate::{
+    key_value_store::*, AssetCode, KeystoreError, Ledger, RecordAmount, TransactionHash,
+    TransactionKind, TransactionReceipt, TransactionUID,
+};
+use chrono::{DateTime, Local};
+use jf_cap::{
+    keys::UserAddress,
+    structs::{ReceiverMemo, RecordOpening},
+    Signature,
+};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::ops::{Deref, DerefMut};
+
+/// A Transaction<L>with its UID as the primary key.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Transaction<L: Ledger> {
```

Comment:
Since we now have a sane way of updating transaction metadata, let's add `status: TransactionStatus`, so that we don't have to do a (confusing) network query to try and guess the status whenever we want it. This will add work to the integration phase, where we will have to update the status whenever a chain event happens (e.g. set it to Rejected when the transaction times out, set it to Accepted when the transaction is finalized) but I think it is an improvement over the horror that is `transaction_status`